### PR TITLE
Add gift line item cart discount

### LINKED ISSUE #1412: Add support for gift cart discount
Our API now support cart discounts that add a gift line item to a cart and we should add support for it to our SDK.

### PR DESCRIPTION
**Pull request by katmatt**
Closes #1412

### CODE REVIEW

**Comment by katmatt** [comment]
@lauraluiz I was unsure to add more integration tests for `Cart#getRefusedGifts()` and decided to just write an unit test for it. WDYT?

**Comment by lauraluiz** [comment]
@katmatt did you read my comments?

File: commercetools-models/src/main/java/io/sphere/sdk/carts/LineItemMode.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Enumerates the available modes for a line item.
+ */
+public enum LineItemMode implements SphereEnumeration {
+    /**
+     * The line item was added during cart creation {@link io.sphere.sdk.carts.commands.CartCreateCommand}
+     * or with the update action {@link io.sphere.sdk.carts.commands.updateactions.AddLineItem}.
+     *
+     * It's quantity can be changed without restrictions.
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/carts/LineItemMode.java]: What does `*lts`mean?

**Comment by katmatt** [comment]
I agree with your opinion with regards to UT vs IT!

**Pull request merged**